common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -419,7 +419,8 @@ protected void wakeup(boolean inEventLoop) {
 
     @Override
     public boolean inEventLoop(Thread thread) {
-        return thread == this.thread;
+        return thread == this.thread ||
+                executor instanceof EventExecutor && ((EventExecutor) executor).inEventLoop(thread);
```

Comment:
I think this is not safe to do as this means you may execute stuff that is only allowed to do by the "real" EventLoop itself. So this opens up all kind of different possible race-conditions.